packages/react-jss/src/createHoc.js

```diff
@@ -137,16 +128,12 @@ export default function createHOC<
       this.unmanage(this.props, this.state)
     }
 
-    get theme() {
-      return isThemingEnabled ? this.props.theme : noTheme
-    }
-
     get jss() {
-      return this.context[ns.jss] || optionsJss || jss
+      return this.props.jss || optionsJss || jss
     }
 
     get manager(): SheetsManager {
-      const managers = this.context[ns.managers]
+      const {managers} = this.props
```

Comment:
why are we using props now instead of context?

Reply:
oh we pass everything from context as props ... hmm this would mean props can override our context.

Reply:
Yeah, but we could just simply put the context after the props.

Reply:
In that case user wouldn't be able to pass certain props to his own component

Reply:
we could pass the context as an own prop, something like `jssContext` and not spread it

Reply:
something is wrong with the way we use context consumer, we should not have that namespace problem

Reply:
we can do something like __jssContext__, but it seems weird

Reply:
I think we can just use `jssContext`

Reply:
Ah actually if you pass it after user props, there is no danger user can override it, so in case yes

Reply:
Done.